test/uq_parameters.jl

```diff
@@ -0,0 +1,224 @@
+# where the new parameteter files resulting from updating an ensemble Kalman
+# ensemble are saved
+@testset "Save parameter ensemble" begin
+
+    # Combine a parameter struct from the parameters defined in
+    # "uq_test_parameters.toml" (the override file) and those defined in
+    # "test_parameters.toml" (the default file)
+    param_path = joinpath(@__DIR__, "test_parameters.toml")
+    uq_param_path = joinpath(@__DIR__,"uq_test_parameters.toml")
+    param_set = CP.create_parameter_struct(
+        uq_param_path,
+        param_path,
+        dict_type="name"
+    )
+
+    # Extract the UQ parameters from the joint set of the parameters from
+    # param_path and those from uq_param_path
+    uq_param_names = CP.get_UQ_parameters(param_set)
+
+    # Seed for pseudo-random number generator
+    rng_seed = 42
+    rng = Random.MersenneTwister(rng_seed)
+    
+    # Construct the parameter distribution
+    pd = CP.get_parameter_distribution(param_set, uq_param_names)
+
+    # ------
+    # Set up an ensemble Kalman process to test writing of parameter ensembles
+    # ------
+
+    # Generate symthetic observations y_obs by evaluating a (completely
+    # contrived) forward map G(u) (where u are the parameters) with the 
+    # the true parameter values u* (which we pretend to know for the
+    # purpose of this example) and adding random observational noise η
+
+    # Define forward map (this is a completely contrived example)
+    A3 = rand([0, 1], 4, 4)
+    A5 = rand([0, 1], 4, 4)
+    function G(u1, u2, u3, u4, u5, u6, u7)  # map from R^5 to R^4
+        A4 = reshape([norm(u7), norm(u6), u1, u2], 4, 1)
+        y = A3 * u3 + A5 * u5 + norm(u4) * A4 
+        return dropdims(y, dims=2) 
+    end
+
+    # True parameter values
+    u1_star = 2.5
+    u2_star = 3.0
+    u3_star = [0.12, -0.05, -0.13, 0.05]
+    u4_star = [4.0, 14.0]
+    u5_star = [2.5, 5.5, 10.0, 14.2]
+    u6_star = ones(10)
+    u7_star = zeros(10)
+    
+    # Synthetic observation
+    y_star = G(u1_star, u2_star, u3_star, u4_star, u5_star, u6_star, u7_star)
+    Γy = 0.05 * I
+    pdf_η = MvNormal(zeros(4), Γy)
+    y_obs = y_star .+ rand(pdf_η)
+
+    N_ens = 50 # number of ensemble members
+    N_iter = 1 # number of iterations
+
+    # Generate and save initial paramter ensemble 
+    initial_ensemble = EKP.construct_initial_ensemble(rng, pd, N_ens)
+    save_path = joinpath(@__DIR__, "test_output")
+    save_file = "test_parameters.toml"
+    CP.save_parameter_ensemble(
+        initial_ensemble,
+        pd,
+        param_set,
+        save_path,
+        save_file,
+        0 # We consider the initial ensemble to be the 0th iteration
+    )
+
+    # Instantiate an ensemble Kalman process
+    eksobj = EKP.EnsembleKalmanProcess(
+        initial_ensemble,
+        y_obs,
+        Γy,
+        Inversion(),
+        rng=rng)
+
+    # EKS iterations
+    for i in 1:N_iter
+        params_i = get_u_final(eksobj)
+        G_n = [G(params_i[1, i],
```

Comment:
see the `batch` function in ParameterDistributions, which should provide the `[1,2,3:6,7:8,...]` etc.

Reply:
Yes, I'll change this - it's a good opportunity to showcase `batch` (of whose existence I didn't know yet when I first wrote that ensemble update :-))!